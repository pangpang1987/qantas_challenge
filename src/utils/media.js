import { css } from "styled-components";

const sizes = {
  screen: 1199,
  desktop: 991,
  tablet: 768,
  phone: 575
};

// Iterate through the sizes and create a media template
export const media = Object.keys(sizes).reduce((acc, label) => {
  acc[label] = (...args) => css`
    @media (max-width: ${sizes[label]}px) {
      ${css(...args)};
    }
  `;
  return acc;
}, {});

export default media;
