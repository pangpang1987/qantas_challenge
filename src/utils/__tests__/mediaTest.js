import media from "../media";

describe("media helper function", () => {
  it("should output a styled css when given proper params", () => {
    const output = media.phone`width: 100%;`;
    expect(output).toMatchSnapshot();
  });
});
