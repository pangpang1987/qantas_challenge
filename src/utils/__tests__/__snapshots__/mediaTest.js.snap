// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media helper function should output a styled css when given proper params 1`] = `
Array [
  "
    @media (max-width: ",
  "575",
  "px) {
      ",
  "width: 100%;",
  ";
    }
  ",
]
`;
